SPECS/postgresql/postgresql.spec

```diff
@@ -101,10 +105,59 @@ chown -Rv nobody .
 sudo -u nobody -s /bin/bash -c "PATH=$PATH make -k check"
 
 %ldconfig_scriptlets
+%pre
+
+if ! getent group postgres >/dev/null; then
+    /sbin/groupadd -r postgres
+fi
+
+if ! getent passwd postgres >/dev/null; then
+    /sbin/useradd -g postgres postgres
+fi
+
+%post
+
+PGDATA="/usr/local/pgsql/data"
+PGRUN="/run/postgresql"
+
+if [ ! -d "$PGDATA" ]; then
+    mkdir -p "$PGDATA"
+    chown postgres:postgres "$PGDATA"
+    sudo -u postgres /usr/bin/initdb -D "$PGDATA"
```

Comment:
I can't recall, but if root is not configured to run sudo commands, ie root ALL(ALL) ALL, won't this fail? 

Would it be better to do
su -c command postgres